src/codemagic/firebase/resource_managers/resource_manager.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC
+from abc import abstractmethod
+from enum import Enum
+from typing import Any
+from typing import Generic
+from typing import List
+from typing import Optional
+from typing import Protocol
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+
+from googleapiclient import discovery
+
+from ..resources.resource import Resource
+
+ResourceT = TypeVar('ResourceT', bound=Resource)
+ParentResourceIdentifierT = TypeVar('ParentResourceIdentifierT', bound='ParentResourceIdentifier')
+
+
+class ParentResourceIdentifier(ABC):
+    @property
+    @abstractmethod
+    def uri(self) -> str:
+        ...
+
+
+class PExecutableInterface(Protocol):
+    def execute(self, **kwargs) -> dict[str, Any]:
```

Comment:
[PEP 585 - Type Hinting Generics In Standard Collections](https://peps.python.org/pep-0585/) was added in Python 3.9. Until then you'd have to use `typting.Dict` for type hinting. And as our codebase is expected to work with Python 3.7, using `dict`, `list` etc literals in type hints is not allowed.